tsnet/tsnet.go

```diff
@@ -92,6 +93,9 @@ func (s *Server) Dial(ctx context.Context, network, address string) (net.Conn, e
 // Start connects the server to the tailnet.
 // Optional: any calls to Dial/Listen will also call Start.
 func (s *Server) Start() error {
+	if runtime.GOOS == "darwin" && runtime.Version() == "go1.18" {
+		log.Fatalf("Tailscale is broken on macOS with go1.18 due to upstream bug https://github.com/golang/go/issues/51759; use 1.18.1+ or Tailscale's Go fork")
```

Comment:
yes, this is a lot.Fatalf in a func returning an error. Meh.

It's better than a panic, which also brought down the whole program! :)

Reply:
Didn't we do something with compile time checks for the go version?

Reply:
You're thinking of 1db46919ab02676faa65535c18da9c45988e3a30 which made Go 1.17 have nice error messages when building HEAD. This is about using Go 1.18(.0) to build anything.  (we'd need to cherry-pick this back)

Reply:
Oh I was wondering if we could detect go 1.18.0 vs go 1.18.X via build tags

This is fine, it's better to have the error at least tell you what's wrong.